### PR TITLE
Add Copilot skills for PR review workflows

### PR DESCRIPTION
**Pull request by laifenxiawucha**
## Summary
Add PR review instructions for copilot and AI Agents.

- copilot-instructions.md : The global hot memory when agents load, every agent will read
- instructions/ : The folder that will be read when reading specific folder
- skills/xpu-ops-pr-review: The skill that will be automatically loaded when doing pr-review

## Testing
- Need to be merged into main , then the review skills will be loaded. Formerly tested in local agent

### CODE REVIEW

File: .claude/skills/skill-writer/SKILL.md
Comment (Stonepia) [inline .claude/skills/skill-writer/SKILL.md:1]: This Skill-writer is good, but I think we should make it in another PR, this PR is for PR Review only

Comment (Stonepia) [reply .claude/skills/skill-writer/SKILL.md:1]: And it's in `.claude`, we don't use claude for now, and we don't have the AGENTS.md, so by default, this file will never be read

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+# PyTorch PR Review Skill
+
+Review PyTorch pull requests focusing on what CI cannot check: code quality, test coverage adequacy, security vulnerabilities, and backward compatibility.
+
+## Usage Modes
+
+### No Argument
+
+If the user invokes `/pr-review` with no arguments, **do not perform a review**. Instead, ask the user what they would like to review:
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: This does not apply to our usecase, right? We assign the copilot as reviewer, so we don't need this.

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+> What would you like me to review?
+> - A PR number or URL (e.g., `/pr-review 12345`)
+> - A local branch (e.g., `/pr-review branch`)
+
+### Local CLI Mode
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: Still don't need

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
+---
+name: pr-review
+description: Review PyTorch pull requests for code quality, test coverage, security, and backward compatibility. Use when reviewing PRs, when asked to review code changes, or when the user mentions "review PR", "code review", or "check this PR".
+---
+
+# PyTorch PR Review Skill
+
+Review PyTorch pull requests focusing on what CI cannot check: code quality, test coverage adequacy, security vulnerabilities, and backward compatibility.
+
+## Usage Modes
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: Overall, I think we don't need this usage mode section. It is redundant, no matter what use case, the LLM will know, you don't need such details

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+```
+
+For local branch reviews:
+- The "Summary" should describe what the branch changes accomplish based on commit messages and diff
+- Use the current branch name in the review header instead of a PR number
+- All other review criteria apply the same as PR reviews
+
+### GitHub Actions Mode
+
+When invoked via `@claude /pr-review` on a GitHub PR, the action pre-fetches PR
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: No, we didn't use this `@claude /pr-review`, and it is not general.

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md (hunk 3 above)
Comment (Stonepia) [reply .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: Suggest to delete this Usage Modes entirely.

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+### Specific Comments
+- `src/module.py:42` - Consider extracting this logic into a named function for clarity
+- `test/test_feature.py:100-105` - Missing test for error case when input is None
+- `torch/nn/modules/linear.py:78` - This allocation could be moved outside the loop
+```
+
+## Files to Reference
+
+When reviewing, consult these project files for context — read them rather than relying on memory, as they change frequently:
+- `CLAUDE.md` - Coding style philosophy and testing patterns
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: We don't have this `CLAUDE.md` for now

File: .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+- `CONTRIBUTING.md` - PR requirements and review process
+- `torch/testing/_internal/common_utils.py` - Test patterns and utilities
+- `torch/testing/_internal/opinfo/core.py` - OpInfo test framework
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/pytorch-pr-review-skill.md]: The `opinfo/core.py` is not needed

File: .github/skills/xpu-ops-pr-review/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+- Writing file-scoped findings with concrete evidence
+- Preparing a repo-ready AI-assisted review comment in English
+
+Do not use this Skill when:
+
+- The task is issue filing or issue-template formatting
+- The task is general XPU debugging without a concrete review context
+- The task is automatic GitHub posting or workflow automation
+
+## Quick Start
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/SKILL.md]: This section not needed.

File: .github/skills/xpu-ops-pr-review/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+Typical prompts that should trigger this Skill:
+
+- "Review this torch-xpu-ops PR"
+- "Check whether this review comment is valid"
+- "Draft a reply to this inline reviewer concern"
+- "Review this branch against main"
+- "Summarize the blocking findings in this PR"
+
+## Load First
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/SKILL.md]: This section is too big! If you load all these references at the beginning, then the context window would be too long at init.

File: .github/skills/xpu-ops-pr-review/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+
+## Review Modes
+
+- Default mode is detailed PR review mode.
+- Detailed PR review mode means the final answer is file-scoped, evidence-based, and anchored to the smallest verified line, hunk, function, registration block, or test block.
+- Thread reply mode is for answering one existing PR comment or inline review thread with a concise maintainer-style reply.
+- Quick triage mode is for identifying whether a concern looks valid before drafting a full reply.
+- Only collapse to a brief summary mode when the human explicitly asks for a quick, concise, or high-level review.
+
+## Review Philosophy
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/SKILL.md]: This is duplicated with https://github.com/intel/torch-xpu-ops/pull/3364/changes#diff-371168fc114a85f50f53b0333ab44fcf874434164a8dbb279e77605f27301490R114

I suggest that you put the `xpu-ops-pr-review/SKILL.md` to only a small context map, and define the workflow steps. Then separate each into files.

File: .github/copilot-instructions.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+This repository implements Torch XPU Operators for Intel GPU support.
+
+Repository layout:
+- `src/` contains XPU operator implementations and backend-specific code.
+- `yaml/` contains operator/config definitions and generated-input sources.
+- `test/` contains regression, unit, and behavior validation.
+- `.ci/` contains validation and CI-related workflows.
+- `tools/` contains support scripts and utilities.
+
+When reviewing pull requests in this repository:
```
Comment (Stonepia) [inline .github/copilot-instructions.md]: From the sections below, the reviewing should not be put into copilot-instructions, you should ask copilot to load the skills under the SKILLS folder instead

File: .github/instructions/xpu-kernels.instructions.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
+---
+applyTo: "src/**,yaml/**"
```
[the remark this answers is not in the document]
Comment (Stonepia) [reply .github/instructions/xpu-kernels.instructions.md]: @laifenxiawucha Please fix this, this seems reasonable

[the remark this answers is not in the document]
Comment (laifenxiawucha) [reply .github/instructions/xpu-kernels.instructions.md]: Replying to [this mention](https://github.com/intel/torch-xpu-ops/pull/3364#discussion_r3098060227).

Could you clarify what fix you're asking for? The mention says "Please fix this, this seems reasonable" but I'm not sure which part of the PR needs fixing. Are you pointing to a specific line in the diff, or is there a particular issue you'd like addressed?

File: .github/skills/xpu-ops-pr-review/references/torch-xpu-ops-review-notes.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+- `src/ATen/native/xpu/`
+- `src/ATen/native/xpu/sycl/`
+- `src/ATen/native/xpu/XPUFallback.template`
+- `test/xpu/`
+- `test/test_ops_xpu.py`
+- `.github/workflows/` when the PR changes CI, packaging, or platform coverage
+
+## Eight High-Risk Review Areas
+
+### 1. Semantic Parity Must Match CPU Or CUDA Behavior
```
Comment (Stonepia) [inline .github/skills/xpu-ops-pr-review/references/torch-xpu-ops-review-notes.md:38]: The CPU/CUDA code are in pytorch repo, so you need to explicitly tell the agent to clone/fetch the pytorch source for reference, otherwise, it will infer by LLM, leading to possible bad result.

**Review by Stonepia — APPROVED** [review]
LGTM

**Review by EikanWang — APPROVED** [review]

**Comment by chuanqi129** [comment]
✅ PR has been successfully merged by @chuanqi129 (force merge).
